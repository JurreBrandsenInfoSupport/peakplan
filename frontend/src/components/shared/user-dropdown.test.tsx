import { expect, describe, it, vi } from "vitest";
import { render } from "@testing-library/react";
import UserDropdown from "./user-dropdown";
import { SidebarProvider } from "../ui/sidebar";

// Mock the next/navigation module
const pushMock = vi.fn();
vi.mock("next/navigation", () => ({
  useRouter: () => ({
    push: pushMock,
  }),
}));

// Mock the use-mobile hook to avoid mobile detection issues in tests
vi.mock("@/hooks/use-mobile", () => ({
  useIsMobile: () => false,
}));

const renderWithProviders = (component: React.ReactNode) => {
  return render(<SidebarProvider>{component}</SidebarProvider>);
};

describe("UserDropdown", () => {
  it("renders correctly", () => {
    const component = renderWithProviders(<UserDropdown />);
    expect(component).toMatchSnapshot();
  });
});
