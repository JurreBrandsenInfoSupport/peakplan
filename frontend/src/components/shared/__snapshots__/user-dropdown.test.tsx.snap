// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDropdown > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
        data-slot="sidebar-wrapper"
        style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
      >
        <ul
          class="flex w-full min-w-0 flex-col gap-1"
          data-sidebar="menu"
          data-slot="sidebar-menu"
        >
          <li
            class="group/menu-item relative"
            data-sidebar="menu-item"
            data-slot="sidebar-menu-item"
          >
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-hidden ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-data-[sidebar=menu-action]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:size-8! group-data-[collapsible=icon]:p-2! [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm"
              data-active="false"
              data-sidebar="menu-button"
              data-size="default"
              data-slot="dropdown-menu-trigger"
              data-state="closed"
              id="radix-«r0»"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-user-round"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="8"
                  r="5"
                />
                <path
                  d="M20 21a8 8 0 0 0-16 0"
                />
              </svg>
              <span>
                Current User
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-up ml-auto"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m18 15-6-6-6 6"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
      data-slot="sidebar-wrapper"
      style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
    >
      <ul
        class="flex w-full min-w-0 flex-col gap-1"
        data-sidebar="menu"
        data-slot="sidebar-menu"
      >
        <li
          class="group/menu-item relative"
          data-sidebar="menu-item"
          data-slot="sidebar-menu-item"
        >
          <button
            aria-expanded="false"
            aria-haspopup="menu"
            class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-hidden ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-data-[sidebar=menu-action]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:size-8! group-data-[collapsible=icon]:p-2! [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm"
            data-active="false"
            data-sidebar="menu-button"
            data-size="default"
            data-slot="dropdown-menu-trigger"
            data-state="closed"
            id="radix-«r0»"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-user-round"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="8"
                r="5"
              />
              <path
                d="M20 21a8 8 0 0 0-16 0"
              />
            </svg>
            <span>
              Current User
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-up ml-auto"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m18 15-6-6-6 6"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
